tests/test_sequence_sampler.py

```diff
@@ -188,6 +209,41 @@ def test_SLM_against_simulation(seq_with_SLM):
     assert_same_samples_as_sim(seq_with_SLM)
 
 
+def test_samples_repr(seq_rydberg):
```

Comment:
It feels really wrong we have to test for this.

Reply:
Why? I think it makes sense it is touched and it's giving the output we expect.

Reply:
I wrote this __repr__  to help me debug by visual inspection, not for actually printing this info somewhere since I doubt thousands of numerical samples are informative.
Now We have to check against an "expected" output format, while I just made it on the fly.
So my point was that's not at all a hard design constraint and with this test it now becomes one 😅

So I am unsure now, maybe we could just remove the __repr__ here, but now that we have it why not keeping it. I'll let you judge!

Reply:
I see. I think that now it's there, it doesn't hurt, so I'll just keep it